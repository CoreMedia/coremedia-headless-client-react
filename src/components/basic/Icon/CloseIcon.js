// @flow
import React from 'react';
import PropTypes from 'prop-types';

import Svg from './Svg';

type Props = {
  color?: string,
  size?: number,
};

const CloseIcon = ({ color = 'currentColor', size }: Props) => {
  return (
    <Svg title="close" size={size}>
      <path
        fill={color}
        d="M256 0C114.624 0 0 114.624 0 256s114.624 256 256 256 256-114.624 256-256S397.376 0 256 0zm126.624 337.376l-45.248 45.248L256 301.248l-81.376 81.376-45.248-45.248L210.752 256l-81.376-81.376 45.248-45.248L256 210.752l81.376-81.376 45.248 45.248L301.248 256l81.376 81.376z"
      />
    </Svg>
  );
};

CloseIcon.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
};

export default CloseIcon;
