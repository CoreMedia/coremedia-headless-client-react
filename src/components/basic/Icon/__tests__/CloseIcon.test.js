// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import CloseIcon from '../CloseIcon';

jest.mock('uuid/v4');

describe('CloseIcon Component', () => {
  it('should render correctly with default props', () => {
    const tree = renderer.create(<CloseIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with color #000', () => {
    const tree = renderer.create(<CloseIcon color="#000" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with size 40', () => {
    const tree = renderer.create(<CloseIcon size={40} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
