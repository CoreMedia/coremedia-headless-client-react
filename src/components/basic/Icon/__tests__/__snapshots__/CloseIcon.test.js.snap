// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon Component should render correctly with color #000 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  height: 100%;
}

<svg
  aria-labelledby="ID"
  className="c0"
  role="img"
  size={undefined}
  viewBox="0 0 512 512"
>
  <title
    id="ID"
  >
    close
  </title>
  <path
    d="M256 0C114.624 0 0 114.624 0 256s114.624 256 256 256 256-114.624 256-256S397.376 0 256 0zm126.624 337.376l-45.248 45.248L256 301.248l-81.376 81.376-45.248-45.248L210.752 256l-81.376-81.376 45.248-45.248L256 210.752l81.376-81.376 45.248 45.248L301.248 256l81.376 81.376z"
    fill="#000"
  />
</svg>
`;

exports[`CloseIcon Component should render correctly with default props 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  height: 100%;
}

<svg
  aria-labelledby="ID"
  className="c0"
  role="img"
  size={undefined}
  viewBox="0 0 512 512"
>
  <title
    id="ID"
  >
    close
  </title>
  <path
    d="M256 0C114.624 0 0 114.624 0 256s114.624 256 256 256 256-114.624 256-256S397.376 0 256 0zm126.624 337.376l-45.248 45.248L256 301.248l-81.376 81.376-45.248-45.248L210.752 256l-81.376-81.376 45.248-45.248L256 210.752l81.376-81.376 45.248 45.248L301.248 256l81.376 81.376z"
    fill="currentColor"
  />
</svg>
`;

exports[`CloseIcon Component should render correctly with size 40 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  width: 40px;
  height: 40px;
}

<svg
  aria-labelledby="ID"
  className="c0"
  role="img"
  size={40}
  viewBox="0 0 512 512"
>
  <title
    id="ID"
  >
    close
  </title>
  <path
    d="M256 0C114.624 0 0 114.624 0 256s114.624 256 256 256 256-114.624 256-256S397.376 0 256 0zm126.624 337.376l-45.248 45.248L256 301.248l-81.376 81.376-45.248-45.248L210.752 256l-81.376-81.376 45.248-45.248L256 210.752l81.376-81.376 45.248 45.248L301.248 256l81.376 81.376z"
    fill="currentColor"
  />
</svg>
`;
